### ISSUE TITLE
overloaded default method of a Protocol errors on subclass.

### ISSUE THREAD

**Issue by dalejung** [microsoft/pyright#4083]
**Describe the bug**
`typing.overload`ed default methods aren't properly inherited by subclass. Pyright thinks the subclass should implement the method.

**To Reproduce**
```python
from typing import Protocol, overload


class BarkProtocol(Protocol):
    @overload
    def bark(self, message: int) -> int:
        ...

    @overload
    def bark(self, message: str) -> str:
        ...

    def bark(self, message: str | int):
        return message


class Dog(BarkProtocol):
    pass
```
gets me the error:

```
error: Class derives from one or more protocol classes but does not implement all required members
    Member "bark" is declared in protocol class "BarkProtocol" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations
```

**Expected behavior**
This should not error since the Protocol provides the default implementation. `mypy` does not error.

Interestingly, if you replace the body of the overload with `pass`, pyright does not register there being an error.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.277, which I just published. It will also be included in a future release of pylance.
